tests/testControlsClass.m

```diff
@@ -342,5 +342,240 @@ function testDisplayGroups(testCase)
+            testCase.verifyEqual(testCase.controls.boundHandling, boundHandlingOptions.Fold.value, 'setProcedure method is not working');
+            testCase.verifyEqual(testCase.controls.parallel, parallelOptions.Single.value, 'setProcedure method is not working');
+            testCase.verifyEqual(testCase.controls.calcSldDuringFit, false, 'setProcedure method is not working');
+            testCase.verifyEqual(testCase.controls.resamPars, [0.9 50], 'setProcedure method is not working');
+            testCase.verifyEqual(testCase.controls.display, displayOptions.Iter.value, 'setProcedure method is not working');
+
+            % Test passing valid parameter values for dream procedure
+            testCase.controls = testCase.controls.setProcedure(procedures.Dream.value,...
+                {'nSamples', 70000,...
+                'nChains', 15,...
+                'lambda', 0.8,...
+                'pUnitGamma', 0.5, ...
+                'boundHandling', boundHandlingOptions.Reflect.value,...
+                'parallel', parallelOptions.Contrasts.value,...
+                'calcSldDuringFit', true,...
+                'resamPars', [0 10],...
+                'display', displayOptions.Notify.value});
+            testCase.verifyEqual(testCase.controls.procedure, procedures.Dream.value, 'setProcedure method is not working');
+            testCase.verifyEqual(testCase.controls.nSamples, 70000, 'setProcedure method is not working');
+            testCase.verifyEqual(testCase.controls.nChains, 15, 'setProcedure method is not working');
+            testCase.verifyEqual(testCase.controls.lambda, 0.8, 'setProcedure method is not working');
+            testCase.verifyEqual(testCase.controls.pUnitGamma, 0.5, 'setProcedure method is not working');
+            testCase.verifyEqual(testCase.controls.boundHandling, boundHandlingOptions.Reflect.value, 'setProcedure method is not working');
+            testCase.verifyEqual(testCase.controls.parallel, parallelOptions.Contrasts.value, 'setProcedure method is not working');
+            testCase.verifyEqual(testCase.controls.calcSldDuringFit, true, 'setProcedure method is not working');
+            testCase.verifyEqual(testCase.controls.resamPars, [0 10], 'setProcedure method is not working');
+            testCase.verifyEqual(testCase.controls.display, displayOptions.Notify.value, 'setProcedure method is not working');
+
+            % Test passing invalid parameter values for dream procedurE
+            testCase.verifyError(@() testCase.controls.setProcedure(procedures.Dream.value,...
+                {'boundHandling', 'invalid'}), invalidOption.errorID);
+            
+            % Test passing wrong parameter for dream procedure
+            testCase.verifyError(@() testCase.controls.setProcedure(procedures.Dream.value,...
+                {'tolX', 1e-6}), invalidParameter.errorID); % Simplex Parameter
```

Comment:
I'm not sure of the need for a new custom exception here - I think this fits into the invalidOption exception, which covers possible options that are not valid in a particular case.